api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/AvailabilityHandler.java

```diff
@@ -0,0 +1,335 @@
+            return Futures.transform(future, ApiUtils.MAP_VOID);
+        });
+    }
+
+    @GET
+    @Path("/{tenantId}/availability")
+    @ApiOperation(value = "Find availabilities metrics data by their tags.", response = Map.class,
+        responseContainer = "List")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successfully fetched data."),
+            @ApiResponse(code = 204, message = "No matching data found."),
+            @ApiResponse(code = 400, message = "Missing or invalid tags query", response = ApiError.class),
+            @ApiResponse(code = 500, message = "Any error in the query.", response = ApiError.class), })
+    public void findAvailabilityDataByTags(@Suspended final AsyncResponse asyncResponse,
+            @PathParam("tenantId") String tenantId,
+            @ApiParam(value = "Tag list", required = true) @QueryParam("tags") Tags tags) {
+        executeAsync(asyncResponse, () -> {
+            if (tags == null) {
+                return badRequest(new ApiError("Missing tags query"));
+            }
+            ListenableFuture<Map<MetricId, Set<Availability>>> future;
+            future = metricsService.findAvailabilityByTags(tenantId, tags.getTags());
+            return Futures.transform(future, ApiUtils.MAP_MAP);
+        });
+    }
+
+    @GET
+    @Path("/{tenantId}/availability/{id}/data")
+    @ApiOperation(value = "Retrieve availability data. When buckets or bucketDuration query parameter is used, "
+            + "the time range between start and end will be divided in buckets of equal duration, and availability "
+            + "statistics will be computed for each bucket.", response = List.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully fetched availability data."),
+            @ApiResponse(code = 204, message = "No availability data was found."),
+            @ApiResponse(code = 400, message = "buckets or bucketDuration parameter is invalid, or both are used.",
+                response = ApiError.class),
+            @ApiResponse(code = 500, message = "Unexpected error occurred while fetching availability data.",
+                response = ApiError.class), })
+    public void findAvailabilityData(
+            @Suspended AsyncResponse asyncResponse,
+            @PathParam("tenantId") String tenantId,
+            @PathParam("id") String id,
+            @ApiParam(value = "Defaults to now - 8 hours") @QueryParam("start") final Long start,
+            @ApiParam(value = "Defaults to now") @QueryParam("end") final Long end,
+            @ApiParam(value = "Total number of buckets") @QueryParam("buckets") Integer bucketsCount,
+            @ApiParam(value = "Bucket duration") @QueryParam("bucketDuration") Duration bucketDuration,
+            @ApiParam(value = "Set to true to return only distinct, contiguous values")
+                @QueryParam("distinct") Boolean distinct) {
+        executeAsync(
```

Comment:
They are long functions; however, I tend to think it is somewhat outside the scope of this PR because that is how the functions already exist in master. I agree that they should be refactored, but I think it should be a separate PR.